### PR TITLE
Do not override already existing default_value with default_option.

### PR DESCRIPTION
**Pull request by KarinG**
Overview
----------------------------------------
Using token module we can set defaults like this:

http://d8civicrm.local/form/paid-membership?membership=3

image

Before
----------------------------------------
`#default_value': '[current-page:query:membership]'`
had to be added in Source as the value would not save when Editing the Membership - User Select - Element

After
----------------------------------------
Saving it!

The first commit contains the fix and is composer requiring Token module. I'll be adding a Test.

### CODE REVIEW

**Comment by KarinG** [comment]
Ok calling it a night - I will continue with the actual test tomorrow!

File: src/Plugin/WebformElement/CivicrmSelect.php [hunk 1]
```diff
@@ -77,7 +77,9 @@ public function prepare(array &$element, WebformSubmissionInterface $webform_sub
       }
       $element['#options'] = $new;
     }
-    $element['#default_value'] = $element['#default_option'];
+    if (isset($element['#default_option'])) {
+      $element['#default_value'] = $element['#default_option'];
+    }
   }
```
Comment (KarinG) [inline src/Plugin/WebformElement/CivicrmSelect.php:83]: Ran into this `Notice: Undefined index: #default_option in Drupal\webform_civicrm\Plugin\WebformElement\CivicrmSelect->prepare() (line 80 of /var/www/drupal/web/sites/default/modules/webform_civicrm/src/Plugin/WebformElement/CivicrmSelect.php)`

**Comment by KarinG** [comment]
I'm going to completely refactor this test rather than jamming it in with Free Membership, as really it has little to do with Free Memberships :-) However -> the test must include a CiviCRM Options type of element.

**Comment by KarinG** [comment]
This is from the test -> so far so good:
image

Getting:
`Behat\Mink\Exception\ElementNotFoundException : Element matching css "tabby-toggle_webform-tab--advanced" not found.`

Ah - just click!
`$this->getSession()->getPage()->clickLink('Advanced');`

**Comment by KarinG** [comment]
Ok this is new:
`Drupal\Tests\webform_civicrm\FunctionalJavascript\MembershipSubmissionTest::testSubmitMembershipQueryParams
Exception: element not interactable`

**Comment by KarinG** [comment]
Ah it's in the fieldset -> and it's collapsed:

image

**Comment by KarinG** [comment]
Ok - one more notice -> here on view of the webform:

image

**Comment by KarinG** [comment]
How do we not have a `#form_key`?

Submission of data is fine - I'm going to leave the notices on form view and form submit as a ToDo for now.

**Comment by KarinG** [comment]
I’d like to merge this PR and then follow up with an additional PR that will prevent filling dblog for people using such URL query arguments. Notices are easy to reproduce (see two ToDo statements).

**Comment by KarinG** [comment]
@jitendrapurohit - definitively a 'squash and merge' on this one 🙂

**Comment by KarinG** [comment]
Oh ... deja vu! Must use `rootUser` when going into Edit Element (of type CiviCRM Options) -> else we loose `#form_key` Tests should all pass now: my local is green ✅

**Comment by jitendrapurohit** [comment]
Have tested this and works great 👍. Thanks @KarinG

**Pull request merged**